rocky/rocky/scheduler.py

```diff
@@ -239,29 +237,33 @@ def get_task_details(self, organization_code: str, task_id: str) -> Task:
 
     def push_task(self, queue_name: str, prioritized_item: PrioritizedItem) -> None:
         try:
-            res = self.session.post(f"{self._base_uri}/queues/{queue_name}/push", data=prioritized_item.json())
+            res = self._client.post(
+                f"/queues/{queue_name}/push",
+                content=prioritized_item.json(),
+                headers={"Content-Type": "application/json"},
+            )
             res.raise_for_status()
-        except HTTPError as http_error:
+        except HTTPStatusError as http_error:
             code = http_error.response.status_code
-            if code == HTTPStatus.TOO_MANY_REQUESTS:
+            if code == codes.TOO_MANY_REQUESTS:
                 raise TooManyRequestsError()
-            elif code == HTTPStatus.BAD_REQUEST:
+            elif code == codes.BAD_REQUEST:
                 raise BadRequestError()
-            elif code == HTTPStatus.CONFLICT:
+            elif code == codes.CONFLICT:
                 raise ConflictError()
             else:
                 raise SchedulerError()
 
     def health(self) -> ServiceHealth:
-        health_endpoint = self.session.get(f"{self._base_uri}/health")
+        health_endpoint = self._client.get("/health")
         health_endpoint.raise_for_status()
         return ServiceHealth.model_validate_json(health_endpoint.content)
 
     def get_task_stats(self, organization_code: str, task_type: str) -> dict:
         try:
-            res = self.session.get(f"{self._base_uri}/tasks/stats/{task_type}-{organization_code}")
+            res = self._client.get(f"/tasks/stats/{task_type}-{organization_code}")
             res.raise_for_status()
-        except HTTPError:
+        except HTTPStatusError:
```

Comment:
We should catch `RequestError` here.

Reply:
We really need to catch `HTTPError`, because the other rocky code assumes that `ScheduleError` is thrown if there is a problem.